scripts/build/TestImpactAnalysis/tiaf.py

```diff
@@ -18,16 +18,227 @@
+        @param args: The arguments to be parsed and applied to this TestImpact object.
         """
 
+        self._runtime_args = []
+        self._change_list = {"createdFiles": [], "updatedFiles": [], "deletedFiles": []}
         self._has_change_list = False
-        self._parse_config_file(config_file)
+        self._use_test_impact_analysis = False
+
+        # Unique instance id to be used as part of the report name.
+        self._instance_id = uuid.uuid4().hex
+
+        self._s3_bucket = args.get('s3_bucket')
+        self._suite = args.get('suite')
+
+        self._config = self._parse_config_file(args['config'])
+
+        # Initialise branches
+        self._src_branch = args.get("src_branch")
+        self._dst_branch = args.get("dst_branch")
+        logger.info(f"Src branch: '{self._src_branch}'.")
+        logger.info(f"Dst branch: '{self._dst_branch}'.")
+
+        # Determine our source of truth. Also initialises our source of truth property.
+        self._determine_source_of_truth()
+
+        # Initialise commit info
+        self._dst_commit = args.get("commit")
+        logger.info(f"Commit: '{self._dst_commit}'.")
+        self._src_commit = None
+        self._commit_distance = None
+
+        # Default to a regular sequence run unless otherwise specified
+        sequence_type = "regular"
+
+        # If flag is set for us to use TIAF
+        if self._use_test_impact_analysis:
+            logger.info("Test impact analysis is enabled.")
+            self._persistent_storage = self._initialise_persistent_storage(
+                s3_bucket=self._s3_bucket, suite=self._suite, s3_top_level_dir=args.get('s3_top_level_dir'))
+
+            # If persistent storage initialised correctly
+            if self._persistent_storage:
+
+                # Historic Data Handling:
+                # This flag is used to help handle our corner cases if we have historic data.
+                self._can_rerun_with_instrumentation = True
+                if self._persistent_storage.has_historic_data:
+                    logger.info("Historic data found.")
+                    self._handle_historic_data()
+                else:
+                    logger.info("No historic data found.")
+
+                # Determining our sequence type:
+                if self._has_change_list:
+                    if self._is_source_of_truth_branch:
+                        # Use TIA sequence (instrumented subset of tests) for coverage updating branches so we can update the coverage data with the generated coverage
+                        sequence_type = "tia"
+                    else:
+                        # Use TIA no-write sequence (regular subset of tests) for non coverage updating branche
+                        sequence_type = "tianowrite"
+                        # Ignore integrity failures for non coverage updating branches as our confidence in the
```

Comment:
question: This sentence seems to end abruptly 😅

Reply:
Fixed